roscala/src/test/scala/coop/rchain/rosette/PrimSpec.scala

```diff
@@ -318,4 +322,245 @@ class PrimSpec extends FlatSpec with Matchers {
+  }
+
+  it should "fail for non-RblFloat arguments" in {
+    val newCtxt = ctxt.copy(nargs = 2, argvec = Tuple(2, Ob.NIV))
+    flMax.fn(newCtxt) should be('left)
+  }
+
+  "flAbs" should "correctly return absolute value" in {
+    val newCtxt = ctxt.copy(nargs = 1, argvec = Tuple(RblFloat(-2.1)))
+    flAbs.fn(newCtxt) should be(Right(RblFloat(2.1)))
+  }
+
+  it should "fail for non-RblFloat arguments" in {
+    val newCtxt = ctxt.copy(nargs = 1, argvec = Tuple(1, Ob.NIV))
+    flAbs.fn(newCtxt) should be('left)
+  }
+
+  "flExp" should "Returns Euler's number e raised to the power of input value." in {
+    val newCtxt = ctxt.copy(nargs = 1, argvec = Tuple(RblFloat(2.5)))
+    flExp.fn(newCtxt) should be(Right(RblFloat(math.exp(2.5))))
+  }
+
+  it should "fail for non-RblFloat arguments" in {
+    val newCtxt = ctxt.copy(nargs = 1, argvec = Tuple(Ob.NIV))
+    flExp.fn(newCtxt) should be('left)
+  }
+
+  "flExpt" should "correctly return base-number raised to the power power-number" in {
+    val newCtxt = ctxt.copy(nargs = 2, argvec = Tuple(Tuple(RblFloat(2)), Tuple(RblFloat(.5))))
+    flExpt.fn(newCtxt) should be(Right(RblFloat(math.pow(2, .5))))
+  }
+
+  it should "fail for non-RblFloat arguments" in {
+    val newCtxt = ctxt.copy(nargs = 2, argvec = Tuple(2, Ob.NIV))
+    flExpt.fn(newCtxt) should be('left)
+  }
+
+  "flLog" should "correctly Returns the natural logarithm (base e) of a float number." in {
+    val newCtxt = ctxt.copy(nargs = 1, argvec = Tuple(RblFloat(Math.E)))
+    flLog.fn(newCtxt) should be(Right(RblFloat(1)))
+  }
+
+  it should "fail for non-RblFloat arguments" in {
+    val newCtxt = ctxt.copy(nargs = 1, argvec = Tuple(Ob.NIV))
+    flLog.fn(newCtxt) should be('left)
+  }
+
+  "flLog10" should "correctly return the base 10 logarithm of a double value." in {
+    val newCtxt = ctxt.copy(nargs = 1, argvec = Tuple(RblFloat(100.0)))
+    flLog10.fn(newCtxt) should be(Right(RblFloat(2.0)))
+  }
+
+  it should "fail for  non-RblFloat arguments" in {
+    val newCtxt = ctxt.copy(nargs = 1, argvec = Tuple(Ob.NIV))
+    flLog10.fn(newCtxt) should be('left)
+  }
+
+  "flCeil" should "correctly return ceil(input_value)" in {
+    val newCtxt = ctxt.copy(nargs = 1, argvec = Tuple(RblFloat(2.1)))
+    flCeil.fn(newCtxt) should be(Right(RblFloat(3.0)))
+  }
+
+  it should "fail for  non-RblFloat arguments" in {
+    val newCtxt = ctxt.copy(nargs = 1, argvec = Tuple(Ob.NIV))
+    flCeil.fn(newCtxt) should be('left)
+  }
+
+  "flFloor" should "correctly return floor(input_value)" in {
+    val newCtxt = ctxt.copy(nargs = 1, argvec = Tuple(RblFloat(2.1)))
+    flFloor.fn(newCtxt) should be(Right(RblFloat(2.0)))
+  }
+
+  it should "fail for  non-RblFloat arguments" in {
+    val newCtxt = ctxt.copy(nargs = 1, argvec = Tuple(Ob.NIV))
+    flFloor.fn(newCtxt) should be('left)
+  }
+
+  "flAtan" should "correctly return the arc tangent of a value; the returned angle is in the range -pi/2 through pi/2. " in {
```

Comment:
`return the arc tangent of the input value` and put the explanation about the range into the test case as a comment.